shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -1117,8 +1189,25 @@ public boolean shouldRestoreAndSaveState() {
     return true;
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Be wary of overriding this method when using {@code
```

Comment:
I had trouble understanding what this implies specifically. Feel free to use more sentences and/or use examples as needed to explain things.

Reply:
Let me know if this is clearer.